src/group_theory/sylow.lean

```diff
@@ -552,4 +570,88 @@ lemma normal_of_normalizer_condition (hnc : normalizer_condition G)
 normalizer_eq_top.mp $ normalizer_condition_iff_only_full_group_self_normalizing.mp hnc _ $
   normalizer_normalizer _
 
+open_locale big_operators
+
+/-- If all its sylow groups are normal, then a finite group is isomorphic to the direct product
+of these sylow groups.
+
+This uses `(Σ (p : (card G).factorization.support), sylow p G)` as the type of all
+(non-trivial) sylow groups.
+-/
+noncomputable
+def direct_product_of_normal [fintype G]
+  (hn : ∀ {p : ℕ} [fact p.prime] (P : sylow p G), (↑P : subgroup G).normal) :
+  (Π P : (Σ (p : (card G).factorization.support), sylow p G), (↑(P.2) : subgroup G)) ≃* G :=
+begin
```

Comment:
I’m not sure if this is the best way of expressing “all sylow groups of `G`” (preferrably as a `fintype` for compatibility with `pi_hom`)

Reply:
Yaël Dillies: Can't you make the product run over p, then over sylow p G?

Joachim Breitner: Hmm, I thought “not easily” because I can construct the isomorphism only easily from a single \Pi type.

But maybe the way to go is to start with the double product in the lemma statement, and then, for each p, prove the inner product to be isomorphic to “the” unique Sylow group, and then invoke the lemma that produces the isomorphism.

I would need a lemma that deduces \Pi H ~- \Pi J from ∀ i, H i ~- J i (where ~- is mul_eqiuv). I wonder if we have that.

Joachim Breitner: Ah, we do: https://leanprover-community.github.io/mathlib_docs/data/equiv/mul_add.html#mul_equiv.Pi_congr_right

Joachim Breitner: Yes, that’s nicer, will do :-)